Update Zoxide shell setup for Nushell in README

Rikiub opened the pull request:
**Auto-load** files support was added on **Nushell** [v0.96.0](https://www.nushell.sh/blog/2024-07-23-nushell_0_96_0.html), allowing `nu` files to be **auto-loaded** on shell start. This behavior is documented [here](https://www.nushell.sh/book/configuration.html#configuration-overview).

This **PR** replaces the need of have a separate file and source it *"manually"*. Instead, **Zoxide** completions now can be added with one-line command.

Also, **Zoxide** now requires [v0.96.0+](https://www.nushell.sh/blog/2024-07-23-nushell_0_96_0.html) as minimum  **Nushell** version.

Rikiub committed:
Update Nushell setup instructions on README.md

Rikiub committed:
Update Nushell setup instructions on README.md

ajeetdsouza commented:
Unfortunately, this wouldn't work.

The reason zoxide requires you to add `zoxide init nushell | save -f ~/.zoxide.nu` to your Nushell env file is so that the init script is regenerated every time you start a new shell. The advantage of this approach is that every time you update zoxide, the init script gets updated.

Given how often Nushell introduces breaking changes, it is important to make this process smooth. The only other option would be to expect the user to remember to re-run the init script every time they update zoxide, which would be bad UX.

Rikiub committed:
Revert "Update Nushell setup instructions on README.md"

This reverts commit 959ea819293df9a80fd5fb786e7f15c5529e54c1.

Rikiub committed:
Revert "Update Nushell setup instructions on README.md"

This reverts commit e3470085de937c2c327137093bfe0eb61f11c6b3.

Rikiub committed:
Change Nushell setup (again) in README.md

Rikiub commented:
Thanks for the advice, I made some changes.

Now requires write the command inside `config.nu`, so each time the shell is started, the file `autoload/zoxide.nu` is created.

I think should be enough.

Rikiub committed:
Merge branch 'ajeetdsouza:main' into readme

Rikiub committed:
Merge branch 'main' into readme

Rikiub committed:
Change Nushell section again

I get this weird error on the linter tests:

README.md:245:81 MD013/line-length Line length [Expected: 80; Actual: 110]

Therefore I had to separate the text into two lines.

ajeetdsouza commented:
Hey! One more thing that just occurred to me: what happens to users who don't have any autoload directories?

Rikiub commented:
I don't think casual users would replace autoload directories, by default nushell always set one autoload directory under `~/.config/nushell/autoload` so should be fine. Advanced users at this point should know what they are doing.

My reason to use autoload directories is because is cross-platform. `$nu.user-autoload-dirs.0` will always point to system config directory, so it make things more organized.

In my opinion, the second way is the most easy for new users:

```
mkdir $nu.user-autoload-dirs.0
zoxide init nushell | save -f ($nu.user-autoload-dirs.0)/zoxide.nu
```

---

Tell me if you agree so I can send another commit 👋

ajeetdsouza commented:
The current Starship initialization method would require users to manually re-run that command on every Starship update. If they don't remember to do so, they are likely to run into weird bugs as the CLI and init script go out-of-sync.

The current zoxide setup is just 2 lines long and doesn't seem like it has any disadvantages, so I'm closing this. Thanks!
